src/execution/operator/join/physical_blockwise_nl_join.cpp

```diff
@@ -79,16 +79,28 @@ class BlockwiseNLJoinState : public CachingOperatorState {
 public:
 	explicit BlockwiseNLJoinState(ExecutionContext &context, ColumnDataCollection &rhs,
 	                              const PhysicalBlockwiseNLJoin &op)
-	    : cross_product(rhs), left_outer(IsLeftOuterJoin(op.join_type)), match_sel(STANDARD_VECTOR_SIZE),
+	    : op(op), cross_product(rhs), left_outer(IsLeftOuterJoin(op.join_type)), match_sel(STANDARD_VECTOR_SIZE),
 	      executor(context.client, *op.condition) {
 		left_outer.Initialize(STANDARD_VECTOR_SIZE);
+		ResetMatches();
 	}
 
+	const PhysicalBlockwiseNLJoin &op;
 	CrossProductExecutor cross_product;
 	OuterJoinMarker left_outer;
 	SelectionVector match_sel;
 	ExpressionExecutor executor;
 	DataChunk intermediate_chunk;
+	bool found_match[STANDARD_VECTOR_SIZE];
```

Comment:
FWIW, `std::vector` has a "possibly space-efficient" specialization for [bool](https://en.cppreference.com/w/cpp/container/vector_bool).

(I know it was stack-allocated and this is not the point of this PR, but I'm happy to contribute a patch if you think it is worth it :-)

Reply:
I think that's likely not worth it - it's more space efficient but random access is less efficient. Since this is only 2048 bytes that is allocated once per join I think the more efficient random access is a better trade-off here.